Autotools fixes

ajelenak opened the pull request:
This is PR #202 without the changes in the LZ4 plugin because those changed the plugin's `cd_values[]`.
<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Update autotools configurations, plugin paths, and add test scripts for BZIP2, JPEG, LZ4, and LZF plugins.
> 
>   - **Autoconf Updates**:
>     - Update `AC_PREREQ` to 2.71 in `BZIP2/configure.ac`, `JPEG/configure.ac`, `LZ4/configure.ac`, and `LZF/configure.ac`.
>     - Change `AC_INIT` and `AC_CONFIG_HEADERS` syntax in `BZIP2/configure.ac`, `JPEG/configure.ac`, `LZ4/configure.ac`, and `LZF/configure.ac`.
>   - **Plugin Path Changes**:
>     - Rename `HDF5_PLUGIN_DIR` to `HDF5_PLUGIN_PATH` in `BZIP2/configure.ac`, `JPEG/configure.ac`, `LZ4/configure.ac`, and `LZF/configure.ac`.
>     - Update `plugindir` to `@HDF5_PLUGIN_PATH@` in `BZIP2/src/Makefile.am`, `JPEG/src/Makefile.am`, `LZ4/src/Makefile.am`, and `LZF/src/Makefile.am`.
>   - **Testing and Examples**:
>     - Add `run_tests.sh` to `JPEG/example/Makefile.am`, `LZ4/example/Makefile.am`, and `LZF/example/Makefile.am`.
>     - Modify `check_PROGRAMS` and `TESTS` in `JPEG/example/Makefile.am`, `LZ4/example/Makefile.am`, and `LZF/example/Makefile.am` to include test scripts.
>   - **Miscellaneous**:
>     - Update `libtool.m4` and `ltversion.m4` in `BZIP2/m4` to newer versions.
>     - Adjust `JPEG_QUALITY` to 50 in `JPEG/example/h5ex_d_jpeg.c` and update related test files.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=HDFGroup%2Fhdf5_plugins&utm_source=github&utm_medium=referral)<sup> for da1fab332df0176f2b430dda65eab2afcc835a39. You can [customize](https://app.ellipsis.dev/HDFGroup/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

edhartnett committed:
fixed plugin

edhartnett committed:
fixes

edhartnett committed:
fixes

edhartnett committed:
fixes

edhartnett committed:
restored static

edhartnett committed:
fixing jpeg

edhartnett committed:
fixes

captainkirk99 committed:
switched to HDF5_PLUGIN_PATH

captainkirk99 committed:
fixing JPEG

captainkirk99 committed:
fixing JPEG

captainkirk99 committed:
more path

edhartnett committed:
more fixes

edhartnett committed:
Merge branch 'e1' of github.com:captainkirk99/hdf5_plugins into e1

edhartnett committed:
fix

edhartnett committed:
fix

edhartnett committed:
fix

edhartnett committed:
adding lzf to the mix

edhartnett committed:
fixing LZ4 filter

edhartnett committed:
more path fixes

edhartnett committed:
more fixes

edhartnett committed:
adding run_tests.sh

edhartnett committed:
more fixes

ajelenak committed:
Remove HDF copyright from community maintained file

ajelenak committed:
Restore back LZ4 filter plugin code

The proposed changes need more discussion.

github-actions[bot] committed:
Committing clang-format changes

ajelenak committed:
Switch from %ld to %zu for size_t vars in JPEG/example/h5ex_d_jpeg.c

Co-authored-by: ellipsis-dev[bot] <65095814+ellipsis-dev[bot]@users.noreply.github.com>

ajelenak committed:
Fixed typos

ajelenak commented on JPEG/src/H5Zjpeg.c:
```diff
@@ -282,7 +281,7 @@ H5Z_filter_jpeg(unsigned int flags, size_t cd_nelmts, const unsigned int cd_valu
             goto failed;
         }
-        jpegInfo.err = jpeg_std_error(&jpegErr);
+        jpegInfo.err = jpeg_std_error((struct jpeg_error_mgr *)&jpegErr);
```
This line indeed seems to conflict with the line 273 earlier in the code:https://github.com/HDFGroup/hdf5_plugins/blob/b9af703ade42e8471c3d7eb0deffd7e3792f8a14/JPEG/src/H5Zjpeg.c#L273
I think this one is correct.

ajelenak committed:
Remove duplicate jpeg_std_error() call

ajelenak committed:
Update example JPEG expected output test files

ajelenak committed:
Merge master branch

ajelenak committed:
Merge branch 'master' into autotools-fixes

ajelenak committed:
Update h5ex_d_jpeg.ddl in all places

brtnfld commented:
stale comment in LZ4/configure.ac:
The AC_ARG_WITH help string still says "use it as a place for the large (i.e. > 2 GiB) files created during the large file testing", which was copy-pasted from HDF5 and is wrong for a plugin directory. In LZF this comment was correctly removed; it should be removed in LZ4 as well.

Issue — misleading AC_MSG_NOTICE in LZF/configure.ac:
Line AC_MSG_NOTICE([*** This is a configure for LZ4 ***]) is a pre-existing copy-paste bug but it's still in this PR unchanged. Should be fixed to say LZF.

Issue:
 BZIP2/example/Makefile.am, LZ4/example/Makefile.am, and LZF/example/Makefile.am all reference run_tests.sh in both TESTS and EXTRA_DIST, but no run_tests.sh this file does not exist for BZIP2, LZ4, or LZF.


Issue: JPEG/src/H5Zjpeg.c removed jpegInfo.err assignment:

-       jpegInfo.err = jpeg_std_error(&jpegErr);
+
        /* Now we can initialize the JPEG compression object. */
        jpeg_create_compress(&jpegInfo);

The line jpegInfo.err = jpeg_std_error(&jpegErr) was removed from the compression path immediately before jpeg_create_compress(), which violates the libjpeg API contract.

ALSO, 
 debug artifacts left in source:
The #ifdef JPEG_DEBUG / fprintf(stderr, ...) / #endif blocks (which were clean, conditional debug code) were replaced with commented-out printf("****   ...") calls:

+        /* printf("****   decompressing nbytes: %ld\n", nbytes); */
+            /* printf("****   jpegInfo.output_scanline: %d\n", jpegInfo.output_scanline);	     */
+        /* printf("****    compressing nbytes: %ld cd_nelmts %ld\n", nbytes, cd_nelmts); */
These are development leftovers and should be removed entirely before merging. The original #ifdef JPEG_DEBUG pattern was better — it allowed debug output without code churn.


Issue:
LZF/example/h5ex_d_lzf.c:
-  printf("   Number of parameters is %d with the value %u\n", nelmts, values_out[0]);
+  printf("   Number of parameters is %ld with the value %u\n", nelmts, values_out[0]);
%ld is still wrong for size_t — it should also be %zu, consistent with the JPEG fix. %ld works on most LP64 systems (Linux/macOS 64-bit) but is undefined behavior and fails on LLP64 (Windows 64-bit).

brtnfld reviewed (dismissed):
See comment

brtnfld committed:
Address PR #210 review comments

- Remove commented-out debug printf artifacts from JPEG/src/H5Zjpeg.c
- Fix stale large-file-testing comment in LZ4/configure.ac; update to --with-hdf5-plugin-path
- Fix copy-paste bug: LZF/configure.ac AC_MSG_NOTICE said LZ4, now says LZF
- Fix %ld -> %zu for size_t in LZF/example/h5ex_d_lzf.c
- Add missing run_tests.sh for BZIP2, LZ4, and LZF examples

brtnfld committed:
Pin BLOSC and BLOSC2 git fetch to versioned tag

GIT_BRANCH was set to "main" causing the via-git CI jobs to pick up
unreleased upstream changes that break output comparisons. Pin to the
tagged release matching the tgz version so all three build paths
(via-git, via-tgz, inline) use the same source.

brtnfld committed:
Revert BLOSC v1 git branch back to main

brtnfld committed:
Force static transitive deps when building BLOSC2 via FetchContent

c-blosc2 main (3.0.0.dev) fetches zlib-ng and lz4 via FetchContent.
When BUILD_SHARED_LIBS is ON (inline HDF5 builds), those deps build as
shared DLLs that libh5blosc2 then depends on at runtime. The plugin path
doesn't include those DLLs, so LoadLibrary fails and H5Zfilter_avail
returns false.

Setting BUILD_SHARED_LIBS OFF before FetchContent_MakeAvailable(BLOSC2)
forces all transitive deps to build as static, eliminating the runtime
dependency. The value is restored after to avoid affecting other plugins.

brtnfld committed:
Revert "Revert BLOSC v1 git branch back to main"

This reverts commit 12d41e90c16d796b51ce6edbd95ada707f498b80.

brtnfld committed:
Revert "Force static transitive deps when building BLOSC2 via FetchContent"

This reverts commit 71c5a9d347a5f3145d2bfc50b07cf6b4719460d0.

brtnfld committed:
Revert "Pin BLOSC and BLOSC2 git fetch to versioned tag"

This reverts commit 815efe0f5fbcc07a57e5f2331de105667a855013.

jhendersonHDF commented on BZIP2/configure.ac:
```diff
@@ -23,17 +23,17 @@ AC_PROG_INSTALL
-            [HDF5_PLUGIN_DIR=$with_hdf5_plugin_dir])
-AC_MSG_RESULT($HDF5_PLUGIN_DIR)
-AC_SUBST([HDF5_PLUGIN_DIR])
+HDF5_PLUGIN_PATH=${HDF5_PLUGIN_PATH-'/usr/local/hdf5/lib/plugin'}
```
```suggestion
HDF5_PLUGIN_PATH=${HDF5_PLUGIN_PATH:-'/usr/local/hdf5/lib/plugin'}
```

at least I think that was the intention. Seems to be in several other configure.ac files to be checked.

brtnfld committed:
Use :- instead of - for shell variable default assignment in configure.ac

Use ${VAR:-default} so the default applies when VAR is unset or empty,
not only when unset.

jhendersonHDF reviewed (approved):
Current testing failures are unrelated to the changes and need some CMake work to fix
